QUARKUS-3007.md

```diff
@@ -0,0 +1,100 @@
+# QUARKUS-3007: Aarch64/ARM Support for both native and JVM Quarkus apps
+
+Jira: https://issues.redhat.com/browse/QUARKUS-3007
+
+The goal of this feature is to provide production support in the same scope for aarch64 architecture that RHBQ provides
+for x86-64 architecture.
+
+For the purposes of planning, the following tasks therefore need to be executed (ordered by priority):
+* Same test coverage and certification matrix on OCP on aarch64 for both JVM and native mode as on OCP on x86-64 (see
+  https://issues.redhat.com/browse/QUARKUS-3446)
+* Certification of RHBQ native mode on OCP on aarch64
+* Certification of RHBQ in both JVM and native mode in the bare metal scenarios on aarch64
+
+## Scope of the testing
+
+### General
+* Catch-up on the test coverage on OCP to be same for aarch64 as it is for x86-64
+  * Enabling Serverless tests in `openshift-arm` profile once RH Serverless supports aarch64
+  * Running test services that do not yet support aarch64 in a side OpenShift cluster running on x86-64
+    * This pertains to units to tens of tests across multiple modules 
+* Certification of RHBQ native mode on OCP on aarch64
+  * The full native OpenShift coverage will have to be ran in native mode with the Mandrel native builder container.
+* Certification of both the JVM and native modes in bare metal scenarios will require all the aspects to be tested
+  * we support two JDK versions for JVM mode and one JDK version for native mode per release stream
+  * start-stop TS with RHBQ code starters
+  * start-stop special characters TS
+  * bare metal test suite
+  * Quarkus quickstarts
+
+### Impact on test suites and testing automation
+* Catch-up on the test coverage on OCP to be same for aarch64 as it is for x86-64
+  * This means that the aarch64 OCP testing pipeline would require an additional x86-64 cluster to be deployed in AWS
+    and test services be prepared before the test execution by the test pipeline. The other part of this is that the
+    tests that currently use test services not supported on aarch64 would be moved to standalone test suite module, only
```

Comment:
This can be big change depending on number of such a miscellaneous tests in one module. Can you give me a very rough idea how many such a tests from which modules it will be? If they are all with the `ts.arm.missing.services.excludes` it's going to be mess.

Reply:
I'm tracking the task in https://issues.redhat.com/browse/QQE-258 - there's currently three test services we don't have on aarch64:
* RH AMQ streams - excluded by https://github.com/quarkus-qe/quarkus-test-suite/pull/1148/commits/36aab6dc891603cf36ee546274b38f037f19adb5 - 7 test cases in messaging module
* RH AMQ streams operator - excluded by https://github.com/quarkus-qe/quarkus-test-suite/pull/1148/commits/99feea23f4d23fb184c65ded7259963c6ac2b350 - 1 test case in messaging module
* RH SSO - https://github.com/quarkus-qe/quarkus-test-suite/pull/1148/commits/b995d9fe5c7c1616b6303d0f884e321371e91e14 - 10 test cases in security module

Reply:
that's acceptable amount, thanks